osu.Game/Screens/Edit/Timing/TimingScreen.cs

```diff
@@ -132,6 +133,61 @@ protected override void LoadComplete()
                 }, true);
             }
 
+            protected override void Update()
+            {
+                base.Update();
+
+                trackActivePoint();
+            }
+
+            /// <summary>
+            /// Given the user has selected a control point group, we want to track any group which is
+            /// active at the current point in time which matches the type the user has selected.
+            ///
+            /// So if the user is currently looking at a timing point and seeks into the future, a
+            /// future timing point would be automatically selected if it is now the new "current" point.
+            /// </summary>
+            private void trackActivePoint()
+            {
+                // For simplicity only match on the first type of the active control point.
+                var selectedPointType = selectedGroup.Value?.ControlPoints.FirstOrDefault()?.GetType();
```

Comment:
This simplification can lead to the behaviour desired breaking down sometimes. Take this list of control point groups, for instance:

![image](https://user-images.githubusercontent.com/20418176/171256681-e0d1ab0e-3ca6-4968-943f-3d4ec972cab9.png)

The first four kiai points will track fine, but because the fourth one has a timing point (and it is considered the "first"), after reaching it the view will switch to tracking timing points from that point onwards, which is kind of confusing if you don't already know how the code looks.

Reply:
I think this is okay for now. It's rare to have timing and effect data in the same group. In the future we may restructure to make it impossible.